book/course/cmake.md

```diff
@@ -0,0 +1,702 @@
+# CMake
+
+This chapter will look at using [CMake](https://cmake.org/) to build software.
+
+```{note}
+The steps outlined below are performed on the [ARC4](https://arcdocs.leeds.ac.uk/systems/arc4.html) HPC system where CMake exists as a [loadable module](https://arcdocs.leeds.ac.uk/software/infrastructure/cmake.html).
+```
+
+## Introduction
+
+CMake is a collection of tools for building, testing and packaging software. 
+It is open source and cross platform and supports a variety of different build tools including Make, Microsoft Visual Studio, Apple's Xcode and others.
+
+CMake works by generating build files from a series of CMake configuration files which the relevant build tool can be used to build the software from.
+This in many cases means CMake is used to create a series of Makefiles which allow us to use GNU make to build our software.
+
+## Installation
+
+If you don't have CMake installed you can download the latest version for your operating system from the [CMake website](https://cmake.org/download/).
+
+## Building software with CMake
+
+In the following section we'll explore a small example of building the command line tool [cURL](https://curl.se/) using CMake. 
+cURL is a tool for transferring data using various network protocols and is commonly used for downloading data from the internet.
+
+```{note}
+cURL is already installed on ARC4, but we'll look at installing our own personal version for this tutorial.
+```
+
+### Cloning the cURL source code
+
+To get started we need to download the cURL source code. cURL is available via GitHub so we can clone the source code at the most recent tagged release with the following line.
+
+```bash
+$ git clone --single-branch --branch curl-7_85_0 https://github.com/curl/curl.git
+```
+
+This will download the cURL source code with the history of the associated release. You will likely see a warning message about being in a `detached head state` which you can ignore for this tutorial.
+
+### Configuring with CMake
+
+Now we've downloaded the cURL source code we need to configure it before building using CMake.
+To do this we need to create a `build/` directory that will contain all of the autogenerated build file.
```

Comment:
build/ looks weird.  I'd omit the trailing slash.  Same comment applies in a few places, and with install/.